R/derive_param_clinbenefit.R

```diff
@@ -0,0 +1,256 @@
+#' Derive a Clinical Benefit Parameter
+#'
+#' Add a clinical benefit/disease control parameter to the input dataset.
+#'
+#' @details
+#' Clinical benefit/disease control is first identified for looking for subjects
+#' having objective response, and then derived for subjects that have at least
+#' one evaluable non-PD response assessment after a specified amount of time
+#' from a reference date.
+#'
+#' @param dataset Input dataset. This is the dataset to which the clinical
+#' benefit rate parameter will be added.
+#'
+#'   The variables `PARAMCD`, `AVALC`, `ADT`, and those specified by the `by_vars`
+#'   parameter are expected.
+#'
+#' @param dataset_adsl ADSL dataset used as input for populating `subject_keys`
+#' in input dataset.
+#'
+#'   The variables specified by the `subject_keys` parameter and the `reference_date`
+#'   parameter are expected.
+#'
+#' @param filter_source Filter condition in `dataset` that represents records
+#' for overall disease response assessment for a subject at a given timepoint,
+#' e.g. `PARAMCD == "OVR"` or `PARAMCD == "OVRLRESP"`.
+#'
+#' @param source_resp A `date_source` object specifying the dataset, date variable,
+#' and filter condition used to identify objective response status.
+#'
+#' @param source_pd A `date_source` object specifying the dataset, date variable,
+#' and filter condition used to identify disease progression.
+#'
+#' @param source_datasets  A named list of data sets is expected.
+#'
+#'  The list must contain the names provided by the `dataset_name` field of the
+#'  `admiral::date_source()` objects specified for `source_pd` and `source_resp`.
+#'
+#' @param by_vars A named list returned by `vars()` used together with `subject_keys`
```

Comment:
is this parameter really needed? as we'd always just take earliest date i think. i cant imagine a scenario when this would be useful, but maybe i'm missing something. i'm not even really sure where this date ever ends up being useful in analyses

Reply:
i guess it could get folded into `subject_keys`. i suppose i was thinking if people wanted to do investigator and independent review CBR at the same time they could use this with `PARCAT` or something, but you could accomplish the same with `subject_keys`. i think the date would be useful if duration of clinical benefit was ever of interest, but perhaps that is not common

Reply:
The idea is good but not straight forward to implement.

I think to use `PARCATx` as by variable we would need to extend `filter_pd()` to handle the PD date depending on the by variables and we would need to allow expressions for `set_values_to`. The first seems quite complex and the latter is forbidden by our programming strategy. Therefore I would remove the parameter.

Reply:
will remove